bloom-housing/bloom
# fix: move user export to direct download

# Pull Request Template

## Issue Overview

This PR addresses #issue


## Description

This PR tackles the following issues found during the most recent bug bash:
- User export having incorrect email text. This was solved by moving it from an email export to a direct download by using the same patterns as applications and listings exports.
- If you update your birthday as a public user and leave a field empty it was auto-populating it with a value (e.g 2001 for year and 01 for both month and day). This has been solved by making the field required on the frontend so the proper ui errors appear.
- Going to a submitted application from the user profile page was sometimes throwing errors. This was fixed with a null check on the FormSummaryDetails page.

~Note: This does not address the issue with some applications not getting tied to user accounts. That is still being addressed and will be in a follow-up PR~
I have been unable to reproduce this

## How Can This Be Tested/Reviewed?

The following flows should be tested:
1. Logging in as admin and exporting user csv
2. Logging in as a jurisdictional admin and exporting user csv
3. Create a public profile and try to edit the birthday in the settings tab. Deleting any of the fields and attempting to save should throw an error

## Checklist:


## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.
